File: src/Java/algorithms/treeandgraph/SameTree.java
package algorithms.treeandgraph; 

import java.util.*; 
import lib.TreeNode; 

public class SameTree {
    /* Function: Check if two trees are same. 
     * Solution: Recursively check left and right. 
     */ 
    
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if (p == null) {
            return q == null; 
        }
        if (q == null) {
            return p == null; 
        }
        
        if (p.val != q.val || !isSameTree(p.left, q.left) || !isSameTree(p.right, q.right)) {
            return false; 
        }
        return true; 
    }
} 

File: src/CPP/src/leetcode/SameTree.cpp
/*
 * [Source] https://leetcode.com/problems/same-tree/
 * [Difficulty]: Easy
 * [Tag]: Tree
 * [Tag]: Depth-first Search
 */

#include <iostream>

using namespace std;

// [Solution]: Compare each node, its left and right child.
// [Corner Case]:
class Solution {
};

/* Java solution
https://github.com/ZhenyingZhu/ClassicAlgorithms/blob/master/src/algorithms/treeandgraph/SameTree.java
 */

/* Java solution
public class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
		if(q==null && p==null) return true;
		if(q==null || p==null) return false;
		if(q.val!=p.val) return false;
		boolean lSub=isSameTree(p.left, q.left);
		boolean rSub=isSameTree(p.right, q.right);
		return lSub && rSub;
    }
}
 */

int main() {
    Solution sol;

    return 0;
}
